utils/definitions.py

```diff
@@ -0,0 +1,9 @@
+"""
```

Comment:
Tricky to know where the project root definition function should be, though could be potentially confusing if the project root is found from its relation to a file in the utils directory (i.e. if utils is moved then the project root will be moved). Perhaps the project_root could be just defined where it is defined in the core/app.py?